system/jhtree/ctfile.hpp

```diff
@@ -92,11 +98,12 @@ struct __declspec(novtable) jhtree_decl KeyHdr
     __int64 fileSize; /* fileSize - was once used in the bias calculation e0x */
     short nodeKeyLength; /* key length in intermediate level nodes e8x */
     unsigned short version; /* build version - to be updated if key format changes    eax*/
-    short unused[2]; /* unused ecx */
+    unsigned keyFlags; /* extended version of ktype ecx */
     __int64 blobHead; /* fpos of first blob node f0x */
     __int64 metadataHead; /* fpos of first metadata node f8x */
     __int64 bloomHead; /* fpos of bloom table data, if present 100x */
-    __uint64 partitionFieldMask; /* Bitmap indicating partition keyed fields */
+    __uint64 partitionFieldMask; /* Bitmap indicating partition keyed fields 108x */
+    unsigned getFlags() { return keyFlags; }
```

Comment:
Won't this fail on an old index? Do you need to fall back to returning oldktype if keyFlags is (whatever value we can be sure is in old index files).